[Bug]: 登入 API 查詢 `store_id` 的部份報錯

# 發生了什麼錯誤？

因為 #63 有將 `store` 資料表的賣家 ID 重新命名成 `seller_id`，但 #57 沒有改到，所以報錯

# 報錯的 log 記錄

```shell
                    context.executemany,
                )
    
        if self._echo:
    
            self._log_info(statement)
    
            stats = context._get_cache_stats()
    
            if not self.engine.hide_parameters:
                self._log_info(
                    "[%s] %r",
                    stats,
                    sql_util._repr_params(
                        parameters, batches=10, ismulti=context.executemany
                    ),
                )
            else:
                self._log_info(
                    "[%s] [SQL parameters hidden due to hide_parameters=True]"
                    % (stats,)
                )
    
        evt_handled = False
        try:
            if context.executemany:
                if self.dialect._has_events:
                    for fn in self.dialect.dispatch.do_executemany:
                        if fn(cursor, statement, parameters, context):
                            evt_handled = True
                            break
                if not evt_handled:
                    self.dialect.do_executemany(
                        cursor, statement, parameters, context
                    )
            elif not parameters and context.no_parameters:
                if self.dialect._has_events:
                    for fn in self.dialect.dispatch.do_execute_no_params:
                        if fn(cursor, statement, context):
                            evt_handled = True
                            break
                if not evt_handled:
                    self.dialect.do_execute_no_params(
                        cursor, statement, context
                    )
            else:
                if self.dialect._has_events:
                    for fn in self.dialect.dispatch.do_execute:
                        if fn(cursor, statement, parameters, context):
                            evt_handled = True
                            break
                if not evt_handled:
>                   self.dialect.do_execute(
                        cursor, statement, parameters, context
                    )

../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/engine/base.py:1900: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <sqlalchemy.dialects.postgresql.psycopg2.PGDialect_psycopg2 object at 0x10d548b20>, cursor = <cursor object at 0x10e711c70; closed: -1>
statement = "\n        SELECT users.*, store.id AS store_id\n        FROM users\n        JOIN store\n        ON users.id=store.user_id\n        WHERE users.email='test@gmail.com'\n        "
parameters = {}, context = <sqlalchemy.dialects.postgresql.psycopg2.PGExecutionContext_psycopg2 object at 0x10ec3fa00>

    def do_execute(self, cursor, statement, parameters, context=None):
>       cursor.execute(statement, parameters)
E       psycopg2.errors.UndefinedColumn: column store.user_id does not exist
E       LINE 5:         ON users.id=store.user_id
E                                   ^

../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/engine/default.py:736: UndefinedColumn

The above exception was the direct cause of the following exception:

client = <starlette.testclient.TestClient object at 0x10e595280>

    def test_sign_up_email_already_in_use(client: TestClient):
        json = {
            "email": "test@gmail.com",
            "username": "yu",
            "password": "test",
        }
    
>       response = client.post("/api/auth/signup", json=json)

tests/test_auth.py:192: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/requests/sessions.py:635: in post
    return self.request("POST", url, data=data, json=json, **kwargs)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/testclient.py:473: in request
    return super().request(
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/requests/sessions.py:587: in request
    resp = self.send(prep, **send_kwargs)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/requests/sessions.py:701: in send
    r = adapter.send(request, **kwargs)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/testclient.py:267: in send
    raise exc
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/testclient.py:264: in send
    portal.call(self.app, scope, receive, send)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/anyio/from_thread.py:283: in call
    return cast(T_Retval, self.start_task_soon(func, *args).result())
/usr/local/Cellar/python@3.9/3.9.15/Frameworks/Python.framework/Versions/3.9/lib/python3.9/concurrent/futures/_base.py:446: in result
    return self.__get_result()
/usr/local/Cellar/python@3.9/3.9.15/Frameworks/Python.framework/Versions/3.9/lib/python3.9/concurrent/futures/_base.py:391: in __get_result
    raise self._exception
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/anyio/from_thread.py:219: in _call_func
    retval = await retval
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/fastapi/applications.py:270: in __call__
    await super().__call__(scope, receive, send)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/applications.py:124: in __call__
    await self.middleware_stack(scope, receive, send)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/middleware/errors.py:184: in __call__
    raise exc
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/middleware/errors.py:162: in __call__
    await self.app(scope, receive, _send)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/middleware/exceptions.py:75: in __call__
    raise exc
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/middleware/exceptions.py:64: in __call__
    await self.app(scope, receive, sender)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/fastapi/middleware/asyncexitstack.py:21: in __call__
    raise e
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/fastapi/middleware/asyncexitstack.py:18: in __call__
    await self.app(scope, receive, send)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/routing.py:680: in __call__
    await route.handle(scope, receive, send)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/routing.py:275: in handle
    await self.app(scope, receive, send)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/routing.py:65: in app
    response = await func(request)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/fastapi/routing.py:231: in app
    raw_response = await run_endpoint_function(
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/fastapi/routing.py:162: in run_endpoint_function
    return await run_in_threadpool(dependant.call, **values)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/concurrency.py:41: in run_in_threadpool
    return await anyio.to_thread.run_sync(func, *args)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/anyio/to_thread.py:31: in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/anyio/_backends/_asyncio.py:937: in run_sync_in_worker_thread
    return await future
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/anyio/_backends/_asyncio.py:867: in run
    result = context.run(func, *args)
app/api/auth/router.py:15: in sign_up
    db_user = get_user_by_email(db, user.email)
app/api/auth/service.py:20: in get_user_by_email
    return database.execute(statement).first()
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/orm/session.py:1714: in execute
    result = conn._execute_20(statement, params or {}, execution_options)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/engine/base.py:1705: in _execute_20
    return meth(self, args_10style, kwargs_10style, execution_options)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/sql/elements.py:334: in _execute_on_connection
    return connection._execute_clauseelement(
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/engine/base.py:1572: in _execute_clauseelement
    ret = self._execute_context(
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/engine/base.py:1943: in _execute_context
    self._handle_dbapi_exception(
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/engine/base.py:2124: in _handle_dbapi_exception
    util.raise_(
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/util/compat.py:210: in raise_
    raise exception
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/engine/base.py:1900: in _execute_context
    self.dialect.do_execute(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <sqlalchemy.dialects.postgresql.psycopg2.PGDialect_psycopg2 object at 0x10d548b20>, cursor = <cursor object at 0x10e711c70; closed: -1>
statement = "\n        SELECT users.*, store.id AS store_id\n        FROM users\n        JOIN store\n        ON users.id=store.user_id\n        WHERE users.email='test@gmail.com'\n        "
parameters = {}, context = <sqlalchemy.dialects.postgresql.psycopg2.PGExecutionContext_psycopg2 object at 0x10ec3fa00>

    def do_execute(self, cursor, statement, parameters, context=None):
>       cursor.execute(statement, parameters)
E       sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedColumn) column store.user_id does not exist
E       LINE 5:         ON users.id=store.user_id
E                                   ^
E       
E       [SQL: 
E               SELECT users.*, store.id AS store_id
E               FROM users
E               JOIN store
E               ON users.id=store.user_id
E               WHERE users.email='test@gmail.com'
E               ]
E       (Background on this error at: https://sqlalche.me/e/14/f405)

../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/engine/default.py:736: ProgrammingError
_______________________________________________________________________________________________ test_login _______________________________________________________________________________________________

self = <sqlalchemy.engine.base.Connection object at 0x10df910d0>, dialect = <sqlalchemy.dialects.postgresql.psycopg2.PGDialect_psycopg2 object at 0x10d548b20>
constructor = <bound method DefaultExecutionContext._init_compiled of <class 'sqlalchemy.dialects.postgresql.psycopg2.PGExecutionContext_psycopg2'>>
statement = "\n        SELECT users.*, store.id AS store_id\n        FROM users\n        JOIN store\n        ON users.id=store.user_id\n        WHERE users.email='test@gmail.com'\n        "
parameters = {}, execution_options = immutabledict({'autocommit': symbol('PARSE_AUTOCOMMIT'), 'future_result': True})
args = (<sqlalchemy.dialects.postgresql.psycopg2.PGCompiler_psycopg2 object at 0x10ec3ffd0>, [{}], <sqlalchemy.sql.elements.TextClause object at 0x10df978b0>, [])
kw = {'cache_hit': symbol('CACHE_HIT')}, branched = <sqlalchemy.engine.base.Connection object at 0x10df910d0>, yp = None, conn = <sqlalchemy.pool.base._ConnectionFairy object at 0x10df911c0>
context = <sqlalchemy.dialects.postgresql.psycopg2.PGExecutionContext_psycopg2 object at 0x10df91250>, cursor = <cursor object at 0x10db67d60; closed: -1>, evt_handled = False

    def _execute_context(
        self,
        dialect,
        constructor,
        statement,
        parameters,
        execution_options,
        *args,
        **kw
    ):
        """Create an :class:`.ExecutionContext` and execute, returning
        a :class:`_engine.CursorResult`."""
    
        branched = self
        if self.__branch_from:
            # if this is a "branched" connection, do everything in terms
            # of the "root" connection, *except* for .close(), which is
            # the only feature that branching provides
            self = self.__branch_from
    
        if execution_options:
            yp = execution_options.get("yield_per", None)
            if yp:
                execution_options = execution_options.union(
                    {"stream_results": True, "max_row_buffer": yp}
                )
    
        try:
            conn = self._dbapi_connection
            if conn is None:
                conn = self._revalidate_connection()
    
            context = constructor(
                dialect, self, conn, execution_options, *args, **kw
            )
        except (exc.PendingRollbackError, exc.ResourceClosedError):
            raise
        except BaseException as e:
            self._handle_dbapi_exception(
                e, util.text_type(statement), parameters, None, None
            )
    
        if (
            self._transaction
            and not self._transaction.is_active
            or (
                self._nested_transaction
                and not self._nested_transaction.is_active
            )
        ):
            self._invalid_transaction()
    
        elif self._trans_context_manager:
            TransactionalContext._trans_ctx_check(self)
    
        if self._is_future and self._transaction is None:
            self._autobegin()
    
        context.pre_exec()
    
        if dialect.use_setinputsizes:
            context._set_input_sizes()
    
        cursor, statement, parameters = (
            context.cursor,
            context.statement,
            context.parameters,
        )
    
        if not context.executemany:
            parameters = parameters[0]
    
        if self._has_events or self.engine._has_events:
            for fn in self.dispatch.before_cursor_execute:
                statement, parameters = fn(
                    self,
                    cursor,
                    statement,
                    parameters,
                    context,
                    context.executemany,
                )
    
        if self._echo:
    
            self._log_info(statement)
    
            stats = context._get_cache_stats()
    
            if not self.engine.hide_parameters:
                self._log_info(
                    "[%s] %r",
                    stats,
                    sql_util._repr_params(
                        parameters, batches=10, ismulti=context.executemany
                    ),
                )
            else:
                self._log_info(
                    "[%s] [SQL parameters hidden due to hide_parameters=True]"
                    % (stats,)
                )
    
        evt_handled = False
        try:
            if context.executemany:
                if self.dialect._has_events:
                    for fn in self.dialect.dispatch.do_executemany:
                        if fn(cursor, statement, parameters, context):
                            evt_handled = True
                            break
                if not evt_handled:
                    self.dialect.do_executemany(
                        cursor, statement, parameters, context
                    )
            elif not parameters and context.no_parameters:
                if self.dialect._has_events:
                    for fn in self.dialect.dispatch.do_execute_no_params:
                        if fn(cursor, statement, context):
                            evt_handled = True
                            break
                if not evt_handled:
                    self.dialect.do_execute_no_params(
                        cursor, statement, context
                    )
            else:
                if self.dialect._has_events:
                    for fn in self.dialect.dispatch.do_execute:
                        if fn(cursor, statement, parameters, context):
                            evt_handled = True
                            break
                if not evt_handled:
>                   self.dialect.do_execute(
                        cursor, statement, parameters, context
                    )

../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/engine/base.py:1900: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <sqlalchemy.dialects.postgresql.psycopg2.PGDialect_psycopg2 object at 0x10d548b20>, cursor = <cursor object at 0x10db67d60; closed: -1>
statement = "\n        SELECT users.*, store.id AS store_id\n        FROM users\n        JOIN store\n        ON users.id=store.user_id\n        WHERE users.email='test@gmail.com'\n        "
parameters = {}, context = <sqlalchemy.dialects.postgresql.psycopg2.PGExecutionContext_psycopg2 object at 0x10df91250>

    def do_execute(self, cursor, statement, parameters, context=None):
>       cursor.execute(statement, parameters)
E       psycopg2.errors.UndefinedColumn: column store.user_id does not exist
E       LINE 5:         ON users.id=store.user_id
E                                   ^

../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/engine/default.py:736: UndefinedColumn

The above exception was the direct cause of the following exception:

client = <starlette.testclient.TestClient object at 0x10ebf8790>

    def test_login(client: TestClient):
        json = {
            "email": "test@gmail.com",
            "password": "a",
        }
    
>       response = client.get("/api/auth/login", json=json)

tests/test_auth.py:252: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/requests/sessions.py:600: in get
    return self.request("GET", url, **kwargs)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/testclient.py:473: in request
    return super().request(
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/requests/sessions.py:587: in request
    resp = self.send(prep, **send_kwargs)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/requests/sessions.py:701: in send
    r = adapter.send(request, **kwargs)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/testclient.py:267: in send
    raise exc
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/testclient.py:264: in send
    portal.call(self.app, scope, receive, send)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/anyio/from_thread.py:283: in call
    return cast(T_Retval, self.start_task_soon(func, *args).result())
/usr/local/Cellar/python@3.9/3.9.15/Frameworks/Python.framework/Versions/3.9/lib/python3.9/concurrent/futures/_base.py:446: in result
    return self.__get_result()
/usr/local/Cellar/python@3.9/3.9.15/Frameworks/Python.framework/Versions/3.9/lib/python3.9/concurrent/futures/_base.py:391: in __get_result
    raise self._exception
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/anyio/from_thread.py:219: in _call_func
    retval = await retval
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/fastapi/applications.py:270: in __call__
    await super().__call__(scope, receive, send)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/applications.py:124: in __call__
    await self.middleware_stack(scope, receive, send)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/middleware/errors.py:184: in __call__
    raise exc
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/middleware/errors.py:162: in __call__
    await self.app(scope, receive, _send)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/middleware/exceptions.py:75: in __call__
    raise exc
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/middleware/exceptions.py:64: in __call__
    await self.app(scope, receive, sender)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/fastapi/middleware/asyncexitstack.py:21: in __call__
    raise e
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/fastapi/middleware/asyncexitstack.py:18: in __call__
    await self.app(scope, receive, send)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/routing.py:680: in __call__
    await route.handle(scope, receive, send)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/routing.py:275: in handle
    await self.app(scope, receive, send)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/routing.py:65: in app
    response = await func(request)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/fastapi/routing.py:231: in app
    raw_response = await run_endpoint_function(
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/fastapi/routing.py:162: in run_endpoint_function
    return await run_in_threadpool(dependant.call, **values)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/concurrency.py:41: in run_in_threadpool
    return await anyio.to_thread.run_sync(func, *args)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/anyio/to_thread.py:31: in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/anyio/_backends/_asyncio.py:937: in run_sync_in_worker_thread
    return await future
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/anyio/_backends/_asyncio.py:867: in run
    result = context.run(func, *args)
app/api/auth/router.py:25: in sing_in
    db_user = get_user_by_email(db, user.email)
app/api/auth/service.py:20: in get_user_by_email
    return database.execute(statement).first()
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/orm/session.py:1714: in execute
    result = conn._execute_20(statement, params or {}, execution_options)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/engine/base.py:1705: in _execute_20
    return meth(self, args_10style, kwargs_10style, execution_options)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/sql/elements.py:334: in _execute_on_connection
    return connection._execute_clauseelement(
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/engine/base.py:1572: in _execute_clauseelement
    ret = self._execute_context(
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/engine/base.py:1943: in _execute_context
    self._handle_dbapi_exception(
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/engine/base.py:2124: in _handle_dbapi_exception
    util.raise_(
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/util/compat.py:210: in raise_
    raise exception
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/engine/base.py:1900: in _execute_context
    self.dialect.do_execute(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <sqlalchemy.dialects.postgresql.psycopg2.PGDialect_psycopg2 object at 0x10d548b20>, cursor = <cursor object at 0x10db67d60; closed: -1>
statement = "\n        SELECT users.*, store.id AS store_id\n        FROM users\n        JOIN store\n        ON users.id=store.user_id\n        WHERE users.email='test@gmail.com'\n        "
parameters = {}, context = <sqlalchemy.dialects.postgresql.psycopg2.PGExecutionContext_psycopg2 object at 0x10df91250>

    def do_execute(self, cursor, statement, parameters, context=None):
>       cursor.execute(statement, parameters)
E       sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedColumn) column store.user_id does not exist
E       LINE 5:         ON users.id=store.user_id
E                                   ^
E       
E       [SQL: 
E               SELECT users.*, store.id AS store_id
E               FROM users
E               JOIN store
E               ON users.id=store.user_id
E               WHERE users.email='test@gmail.com'
E               ]
E       (Background on this error at: https://sqlalche.me/e/14/f405)

../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/engine/default.py:736: ProgrammingError
________________________________________________________________________________________ test_login_invalid_email ________________________________________________________________________________________

self = <sqlalchemy.engine.base.Connection object at 0x10ebffa30>, dialect = <sqlalchemy.dialects.postgresql.psycopg2.PGDialect_psycopg2 object at 0x10d548b20>
constructor = <bound method DefaultExecutionContext._init_compiled of <class 'sqlalchemy.dialects.postgresql.psycopg2.PGExecutionContext_psycopg2'>>
statement = "\n        SELECT users.*, store.id AS store_id\n        FROM users\n        JOIN store\n        ON users.id=store.user_id\n        WHERE users.email='y@gamil.com'\n        "
parameters = {}, execution_options = immutabledict({'autocommit': symbol('PARSE_AUTOCOMMIT'), 'future_result': True})
args = (<sqlalchemy.dialects.postgresql.psycopg2.PGCompiler_psycopg2 object at 0x10ebffd60>, [{}], <sqlalchemy.sql.elements.TextClause object at 0x10ebc8e80>, [])
kw = {'cache_hit': symbol('CACHE_MISS')}, branched = <sqlalchemy.engine.base.Connection object at 0x10ebffa30>, yp = None, conn = <sqlalchemy.pool.base._ConnectionFairy object at 0x10ebff940>
context = <sqlalchemy.dialects.postgresql.psycopg2.PGExecutionContext_psycopg2 object at 0x10ebffb80>, cursor = <cursor object at 0x10e711220; closed: -1>, evt_handled = False

    def _execute_context(
        self,
        dialect,
        constructor,
        statement,
        parameters,
        execution_options,
        *args,
        **kw
    ):
        """Create an :class:`.ExecutionContext` and execute, returning
        a :class:`_engine.CursorResult`."""
    
        branched = self
        if self.__branch_from:
            # if this is a "branched" connection, do everything in terms
            # of the "root" connection, *except* for .close(), which is
            # the only feature that branching provides
            self = self.__branch_from
    
        if execution_options:
            yp = execution_options.get("yield_per", None)
            if yp:
                execution_options = execution_options.union(
                    {"stream_results": True, "max_row_buffer": yp}
                )
    
        try:
            conn = self._dbapi_connection
            if conn is None:
                conn = self._revalidate_connection()
    
            context = constructor(
                dialect, self, conn, execution_options, *args, **kw
            )
        except (exc.PendingRollbackError, exc.ResourceClosedError):
            raise
        except BaseException as e:
            self._handle_dbapi_exception(
                e, util.text_type(statement), parameters, None, None
            )
    
        if (
            self._transaction
            and not self._transaction.is_active
            or (
                self._nested_transaction
                and not self._nested_transaction.is_active
            )
        ):
            self._invalid_transaction()
    
        elif self._trans_context_manager:
            TransactionalContext._trans_ctx_check(self)
    
        if self._is_future and self._transaction is None:
            self._autobegin()
    
        context.pre_exec()
    
        if dialect.use_setinputsizes:
            context._set_input_sizes()
    
        cursor, statement, parameters = (
            context.cursor,
            context.statement,
            context.parameters,
        )
    
        if not context.executemany:
            parameters = parameters[0]
    
        if self._has_events or self.engine._has_events:
            for fn in self.dispatch.before_cursor_execute:
                statement, parameters = fn(
                    self,
                    cursor,
                    statement,
                    parameters,
                    context,
                    context.executemany,
                )
    
        if self._echo:
    
            self._log_info(statement)
    
            stats = context._get_cache_stats()
    
            if not self.engine.hide_parameters:
                self._log_info(
                    "[%s] %r",
                    stats,
                    sql_util._repr_params(
                        parameters, batches=10, ismulti=context.executemany
                    ),
                )
            else:
                self._log_info(
                    "[%s] [SQL parameters hidden due to hide_parameters=True]"
                    % (stats,)
                )
    
        evt_handled = False
        try:
            if context.executemany:
                if self.dialect._has_events:
                    for fn in self.dialect.dispatch.do_executemany:
                        if fn(cursor, statement, parameters, context):
                            evt_handled = True
                            break
                if not evt_handled:
                    self.dialect.do_executemany(
                        cursor, statement, parameters, context
                    )
            elif not parameters and context.no_parameters:
                if self.dialect._has_events:
                    for fn in self.dialect.dispatch.do_execute_no_params:
                        if fn(cursor, statement, context):
                            evt_handled = True
                            break
                if not evt_handled:
                    self.dialect.do_execute_no_params(
                        cursor, statement, context
                    )
            else:
                if self.dialect._has_events:
                    for fn in self.dialect.dispatch.do_execute:
                        if fn(cursor, statement, parameters, context):
                            evt_handled = True
                            break
                if not evt_handled:
>                   self.dialect.do_execute(
                        cursor, statement, parameters, context
                    )

../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/engine/base.py:1900: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <sqlalchemy.dialects.postgresql.psycopg2.PGDialect_psycopg2 object at 0x10d548b20>, cursor = <cursor object at 0x10e711220; closed: -1>
statement = "\n        SELECT users.*, store.id AS store_id\n        FROM users\n        JOIN store\n        ON users.id=store.user_id\n        WHERE users.email='y@gamil.com'\n        "
parameters = {}, context = <sqlalchemy.dialects.postgresql.psycopg2.PGExecutionContext_psycopg2 object at 0x10ebffb80>

    def do_execute(self, cursor, statement, parameters, context=None):
>       cursor.execute(statement, parameters)
E       psycopg2.errors.UndefinedColumn: column store.user_id does not exist
E       LINE 5:         ON users.id=store.user_id
E                                   ^

../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/engine/default.py:736: UndefinedColumn

The above exception was the direct cause of the following exception:

client = <starlette.testclient.TestClient object at 0x10ea35160>

    def test_login_invalid_email(client: TestClient):
        json = {
            "email": "y@gamil.com",
            "password": "test",
        }
    
>       response = client.get("/api/auth/login", json=json)

tests/test_auth.py:273: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/requests/sessions.py:600: in get
    return self.request("GET", url, **kwargs)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/testclient.py:473: in request
    return super().request(
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/requests/sessions.py:587: in request
    resp = self.send(prep, **send_kwargs)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/requests/sessions.py:701: in send
    r = adapter.send(request, **kwargs)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/testclient.py:267: in send
    raise exc
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/testclient.py:264: in send
    portal.call(self.app, scope, receive, send)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/anyio/from_thread.py:283: in call
    return cast(T_Retval, self.start_task_soon(func, *args).result())
/usr/local/Cellar/python@3.9/3.9.15/Frameworks/Python.framework/Versions/3.9/lib/python3.9/concurrent/futures/_base.py:446: in result
    return self.__get_result()
/usr/local/Cellar/python@3.9/3.9.15/Frameworks/Python.framework/Versions/3.9/lib/python3.9/concurrent/futures/_base.py:391: in __get_result
    raise self._exception
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/anyio/from_thread.py:219: in _call_func
    retval = await retval
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/fastapi/applications.py:270: in __call__
    await super().__call__(scope, receive, send)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/applications.py:124: in __call__
    await self.middleware_stack(scope, receive, send)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/middleware/errors.py:184: in __call__
    raise exc
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/middleware/errors.py:162: in __call__
    await self.app(scope, receive, _send)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/middleware/exceptions.py:75: in __call__
    raise exc
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/middleware/exceptions.py:64: in __call__
    await self.app(scope, receive, sender)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/fastapi/middleware/asyncexitstack.py:21: in __call__
    raise e
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/fastapi/middleware/asyncexitstack.py:18: in __call__
    await self.app(scope, receive, send)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/routing.py:680: in __call__
    await route.handle(scope, receive, send)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/routing.py:275: in handle
    await self.app(scope, receive, send)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/routing.py:65: in app
    response = await func(request)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/fastapi/routing.py:231: in app
    raw_response = await run_endpoint_function(
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/fastapi/routing.py:162: in run_endpoint_function
    return await run_in_threadpool(dependant.call, **values)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/concurrency.py:41: in run_in_threadpool
    return await anyio.to_thread.run_sync(func, *args)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/anyio/to_thread.py:31: in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/anyio/_backends/_asyncio.py:937: in run_sync_in_worker_thread
    return await future
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/anyio/_backends/_asyncio.py:867: in run
    result = context.run(func, *args)
app/api/auth/router.py:25: in sing_in
    db_user = get_user_by_email(db, user.email)
app/api/auth/service.py:20: in get_user_by_email
    return database.execute(statement).first()
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/orm/session.py:1714: in execute
    result = conn._execute_20(statement, params or {}, execution_options)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/engine/base.py:1705: in _execute_20
    return meth(self, args_10style, kwargs_10style, execution_options)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/sql/elements.py:334: in _execute_on_connection
    return connection._execute_clauseelement(
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/engine/base.py:1572: in _execute_clauseelement
    ret = self._execute_context(
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/engine/base.py:1943: in _execute_context
    self._handle_dbapi_exception(
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/engine/base.py:2124: in _handle_dbapi_exception
    util.raise_(
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/util/compat.py:210: in raise_
    raise exception
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/engine/base.py:1900: in _execute_context
    self.dialect.do_execute(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <sqlalchemy.dialects.postgresql.psycopg2.PGDialect_psycopg2 object at 0x10d548b20>, cursor = <cursor object at 0x10e711220; closed: -1>
statement = "\n        SELECT users.*, store.id AS store_id\n        FROM users\n        JOIN store\n        ON users.id=store.user_id\n        WHERE users.email='y@gamil.com'\n        "
parameters = {}, context = <sqlalchemy.dialects.postgresql.psycopg2.PGExecutionContext_psycopg2 object at 0x10ebffb80>

    def do_execute(self, cursor, statement, parameters, context=None):
>       cursor.execute(statement, parameters)
E       sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedColumn) column store.user_id does not exist
E       LINE 5:         ON users.id=store.user_id
E                                   ^
E       
E       [SQL: 
E               SELECT users.*, store.id AS store_id
E               FROM users
E               JOIN store
E               ON users.id=store.user_id
E               WHERE users.email='y@gamil.com'
E               ]
E       (Background on this error at: https://sqlalche.me/e/14/f405)

../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/engine/default.py:736: ProgrammingError
______________________________________________________________________________________ test_login_invalid_password _______________________________________________________________________________________

self = <sqlalchemy.engine.base.Connection object at 0x10e0370a0>, dialect = <sqlalchemy.dialects.postgresql.psycopg2.PGDialect_psycopg2 object at 0x10d548b20>
constructor = <bound method DefaultExecutionContext._init_compiled of <class 'sqlalchemy.dialects.postgresql.psycopg2.PGExecutionContext_psycopg2'>>
statement = "\n        SELECT users.*, store.id AS store_id\n        FROM users\n        JOIN store\n        ON users.id=store.user_id\n        WHERE users.email='yu2001@gamil.com'\n        "
parameters = {}, execution_options = immutabledict({'autocommit': symbol('PARSE_AUTOCOMMIT'), 'future_result': True})
args = (<sqlalchemy.dialects.postgresql.psycopg2.PGCompiler_psycopg2 object at 0x10db271c0>, [{}], <sqlalchemy.sql.elements.TextClause object at 0x10ecd94f0>, [])
kw = {'cache_hit': symbol('CACHE_HIT')}, branched = <sqlalchemy.engine.base.Connection object at 0x10e0370a0>, yp = None, conn = <sqlalchemy.pool.base._ConnectionFairy object at 0x10e037100>
context = <sqlalchemy.dialects.postgresql.psycopg2.PGExecutionContext_psycopg2 object at 0x10e037e50>, cursor = <cursor object at 0x10e711130; closed: -1>, evt_handled = False

    def _execute_context(
        self,
        dialect,
        constructor,
        statement,
        parameters,
        execution_options,
        *args,
        **kw
    ):
        """Create an :class:`.ExecutionContext` and execute, returning
        a :class:`_engine.CursorResult`."""
    
        branched = self
        if self.__branch_from:
            # if this is a "branched" connection, do everything in terms
            # of the "root" connection, *except* for .close(), which is
            # the only feature that branching provides
            self = self.__branch_from
    
        if execution_options:
            yp = execution_options.get("yield_per", None)
            if yp:
                execution_options = execution_options.union(
                    {"stream_results": True, "max_row_buffer": yp}
                )
    
        try:
            conn = self._dbapi_connection
            if conn is None:
                conn = self._revalidate_connection()
    
            context = constructor(
                dialect, self, conn, execution_options, *args, **kw
            )
        except (exc.PendingRollbackError, exc.ResourceClosedError):
            raise
        except BaseException as e:
            self._handle_dbapi_exception(
                e, util.text_type(statement), parameters, None, None
            )
    
        if (
            self._transaction
            and not self._transaction.is_active
            or (
                self._nested_transaction
                and not self._nested_transaction.is_active
            )
        ):
            self._invalid_transaction()
    
        elif self._trans_context_manager:
            TransactionalContext._trans_ctx_check(self)
    
        if self._is_future and self._transaction is None:
            self._autobegin()
    
        context.pre_exec()
    
        if dialect.use_setinputsizes:
            context._set_input_sizes()
    
        cursor, statement, parameters = (
            context.cursor,
            context.statement,
            context.parameters,
        )
    
        if not context.executemany:
            parameters = parameters[0]
    
        if self._has_events or self.engine._has_events:
            for fn in self.dispatch.before_cursor_execute:
                statement, parameters = fn(
                    self,
                    cursor,
                    statement,
                    parameters,
                    context,
                    context.executemany,
                )
    
        if self._echo:
    
            self._log_info(statement)
    
            stats = context._get_cache_stats()
    
            if not self.engine.hide_parameters:
                self._log_info(
                    "[%s] %r",
                    stats,
                    sql_util._repr_params(
                        parameters, batches=10, ismulti=context.executemany
                    ),
                )
            else:
                self._log_info(
                    "[%s] [SQL parameters hidden due to hide_parameters=True]"
                    % (stats,)
                )
    
        evt_handled = False
        try:
            if context.executemany:
                if self.dialect._has_events:
                    for fn in self.dialect.dispatch.do_executemany:
                        if fn(cursor, statement, parameters, context):
                            evt_handled = True
                            break
                if not evt_handled:
                    self.dialect.do_executemany(
                        cursor, statement, parameters, context
                    )
            elif not parameters and context.no_parameters:
                if self.dialect._has_events:
                    for fn in self.dialect.dispatch.do_execute_no_params:
                        if fn(cursor, statement, context):
                            evt_handled = True
                            break
                if not evt_handled:
                    self.dialect.do_execute_no_params(
                        cursor, statement, context
                    )
            else:
                if self.dialect._has_events:
                    for fn in self.dialect.dispatch.do_execute:
                        if fn(cursor, statement, parameters, context):
                            evt_handled = True
                            break
                if not evt_handled:
>                   self.dialect.do_execute(
                        cursor, statement, parameters, context
                    )

../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/engine/base.py:1900: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <sqlalchemy.dialects.postgresql.psycopg2.PGDialect_psycopg2 object at 0x10d548b20>, cursor = <cursor object at 0x10e711130; closed: -1>
statement = "\n        SELECT users.*, store.id AS store_id\n        FROM users\n        JOIN store\n        ON users.id=store.user_id\n        WHERE users.email='yu2001@gamil.com'\n        "
parameters = {}, context = <sqlalchemy.dialects.postgresql.psycopg2.PGExecutionContext_psycopg2 object at 0x10e037e50>

    def do_execute(self, cursor, statement, parameters, context=None):
>       cursor.execute(statement, parameters)
E       psycopg2.errors.UndefinedColumn: column store.user_id does not exist
E       LINE 5:         ON users.id=store.user_id
E                                   ^

../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/engine/default.py:736: UndefinedColumn

The above exception was the direct cause of the following exception:

client = <starlette.testclient.TestClient object at 0x10e1ee640>

    def test_login_invalid_password(client: TestClient):
        json = {
            "email": "yu2001@gamil.com",
            "password": "t",
        }
    
>       response = client.get("/api/auth/login", json=json)

tests/test_auth.py:288: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/requests/sessions.py:600: in get
    return self.request("GET", url, **kwargs)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/testclient.py:473: in request
    return super().request(
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/requests/sessions.py:587: in request
    resp = self.send(prep, **send_kwargs)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/requests/sessions.py:701: in send
    r = adapter.send(request, **kwargs)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/testclient.py:267: in send
    raise exc
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/testclient.py:264: in send
    portal.call(self.app, scope, receive, send)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/anyio/from_thread.py:283: in call
    return cast(T_Retval, self.start_task_soon(func, *args).result())
/usr/local/Cellar/python@3.9/3.9.15/Frameworks/Python.framework/Versions/3.9/lib/python3.9/concurrent/futures/_base.py:446: in result
    return self.__get_result()
/usr/local/Cellar/python@3.9/3.9.15/Frameworks/Python.framework/Versions/3.9/lib/python3.9/concurrent/futures/_base.py:391: in __get_result
    raise self._exception
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/anyio/from_thread.py:219: in _call_func
    retval = await retval
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/fastapi/applications.py:270: in __call__
    await super().__call__(scope, receive, send)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/applications.py:124: in __call__
    await self.middleware_stack(scope, receive, send)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/middleware/errors.py:184: in __call__
    raise exc
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/middleware/errors.py:162: in __call__
    await self.app(scope, receive, _send)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/middleware/exceptions.py:75: in __call__
    raise exc
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/middleware/exceptions.py:64: in __call__
    await self.app(scope, receive, sender)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/fastapi/middleware/asyncexitstack.py:21: in __call__
    raise e
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/fastapi/middleware/asyncexitstack.py:18: in __call__
    await self.app(scope, receive, send)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/routing.py:680: in __call__
    await route.handle(scope, receive, send)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/routing.py:275: in handle
    await self.app(scope, receive, send)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/routing.py:65: in app
    response = await func(request)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/fastapi/routing.py:231: in app
    raw_response = await run_endpoint_function(
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/fastapi/routing.py:162: in run_endpoint_function
    return await run_in_threadpool(dependant.call, **values)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/starlette/concurrency.py:41: in run_in_threadpool
    return await anyio.to_thread.run_sync(func, *args)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/anyio/to_thread.py:31: in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/anyio/_backends/_asyncio.py:937: in run_sync_in_worker_thread
    return await future
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/anyio/_backends/_asyncio.py:867: in run
    result = context.run(func, *args)
app/api/auth/router.py:25: in sing_in
    db_user = get_user_by_email(db, user.email)
app/api/auth/service.py:20: in get_user_by_email
    return database.execute(statement).first()
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/orm/session.py:1714: in execute
    result = conn._execute_20(statement, params or {}, execution_options)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/engine/base.py:1705: in _execute_20
    return meth(self, args_10style, kwargs_10style, execution_options)
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/sql/elements.py:334: in _execute_on_connection
    return connection._execute_clauseelement(
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/engine/base.py:1572: in _execute_clauseelement
    ret = self._execute_context(
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/engine/base.py:1943: in _execute_context
    self._handle_dbapi_exception(
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/engine/base.py:2124: in _handle_dbapi_exception
    util.raise_(
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/util/compat.py:210: in raise_
    raise exception
../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/engine/base.py:1900: in _execute_context
    self.dialect.do_execute(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <sqlalchemy.dialects.postgresql.psycopg2.PGDialect_psycopg2 object at 0x10d548b20>, cursor = <cursor object at 0x10e711130; closed: -1>
statement = "\n        SELECT users.*, store.id AS store_id\n        FROM users\n        JOIN store\n        ON users.id=store.user_id\n        WHERE users.email='yu2001@gamil.com'\n        "
parameters = {}, context = <sqlalchemy.dialects.postgresql.psycopg2.PGExecutionContext_psycopg2 object at 0x10e037e50>

    def do_execute(self, cursor, statement, parameters, context=None):
>       cursor.execute(statement, parameters)
E       sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedColumn) column store.user_id does not exist
E       LINE 5:         ON users.id=store.user_id
E                                   ^
E       
E       [SQL: 
E               SELECT users.*, store.id AS store_id
E               FROM users
E               JOIN store
E               ON users.id=store.user_id
E               WHERE users.email='yu2001@gamil.com'
E               ]
E       (Background on this error at: https://sqlalche.me/e/14/f405)

../../.local/share/virtualenvs/niupi-backend-bcX2RF7R/lib/python3.9/site-packages/sqlalchemy/engine/default.py:736: ProgrammingError
======================================================================================== short test summary info =========================================================================================
FAILED tests/test_auth.py::test_sign_up_email_has_strip_whitespaces - sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedColumn) column store.user_id does not exist
FAILED tests/test_auth.py::test_sign_up_username_has_strip_whitespaces - sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedColumn) column store.user_id does not exist
FAILED tests/test_auth.py::test_sign_up - sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedColumn) column store.user_id does not exist
FAILED tests/test_auth.py::test_sign_up_email_already_in_use - sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedColumn) column store.user_id does not exist
FAILED tests/test_auth.py::test_login - sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedColumn) column store.user_id does not exist
FAILED tests/test_auth.py::test_login_invalid_email - sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedColumn) column store.user_id does not exist
FAILED tests/test_auth.py::test_login_invalid_password - sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedColumn) column store.user_id does not exist
```

# 補充截圖

--- a/app/api/auth/service.py
+++ b/app/api/auth/service.py
@@ -12,7 +12,7 @@ def get_user_by_email(database: Session, email: str):
         SELECT users.*, store.id AS store_id
         FROM users
         JOIN store
-        ON users.id=store.user_id
+        ON users.id=store.seller_id
         WHERE users.email='{email}'
         """
     )

